### DISCUSSION TITLE
error in the tachometer example

### DISCUSSION THREAD

**Discussion by m66rob** [khoih-prog/ESP32TimerInterrupt#27] [General]
Arduino:1.8.19 (Windows 10), Tarjeta:"ESP32 Dev Module, Disabled, Default 4MB with spiffs (1.2MB APP/1.5MB SPIFFS), 240MHz (WiFi/BT), QIO, 80MHz, 4MB (32Mb), 921600, None"

In file included from C:\Users\IDROID\Documents\Arduino\libraries\ESP32TimerInterrupt\src/ESP32TimerInterrupt.h:58:0,

                 from C:\Users\IDROID\Desktop\RPM_INTERRUP\RPM_INTERRUP.ino:13:

C:\Users\IDROID\Documents\Arduino\libraries\ESP32TimerInterrupt\src/ESP32TimerInterrupt.hpp: In member function 'bool ESP32TimerInterrupt::setFrequency(const float&, bool (* const&)(void*))':

C:\Users\IDROID\Documents\Arduino\libraries\ESP32TimerInterrupt\src/ESP32TimerInterrupt.hpp:355:100: error: 'timer_isr_callback_add' was not declared in this scope

         timer_isr_callback_add(_timerGroup, _timerIndex, _callback, (void *) (uint32_t) _timerNo, 0);

                                                                                                    ^

C:\Users\IDROID\Documents\Arduino\libraries\ESP32TimerInterrupt\src/ESP32TimerInterrupt.hpp: In member function 'void ESP32TimerInterrupt::detachInterrupt()':

C:\Users\IDROID\Documents\Arduino\libraries\ESP32TimerInterrupt\src/ESP32TimerInterrupt.hpp:397:66: error: 'TIMER_INTR_T0' was not declared in this scope

       timer_group_intr_disable(_timerGroup, (_timerIndex == 0) ? TIMER_INTR_T0 : TIMER_INTR_T1);

                                                                  ^

C:\Users\IDROID\Documents\Arduino\libraries\ESP32TimerInterrupt\src/ESP32TimerInterrupt.hpp:397:82: error: 'TIMER_INTR_T1' was not declared in this scope

       timer_group_intr_disable(_timerGroup, (_timerIndex == 0) ? TIMER_INTR_T0 : TIMER_INTR_T1);

                                                                                  ^

C:\Users\IDROID\Documents\Arduino\libraries\ESP32TimerInterrupt\src/ESP32TimerInterrupt.hpp: In member function 'void ESP32TimerInterrupt::disableTimer()':

C:\Users\IDROID\Documents\Arduino\libraries\ESP32TimerInterrupt\src/ESP32TimerInterrupt.hpp:406:66: error: 'TIMER_INTR_T0' was not declared in this scope

       timer_group_intr_disable(_timerGroup, (_timerIndex == 0) ? TIMER_INTR_T0 : TIMER_INTR_T1);

                                                                  ^

C:\Users\IDROID\Documents\Arduino\libraries\ESP32TimerInterrupt\src/ESP32TimerInterrupt.hpp:406:82: error: 'TIMER_INTR_T1' was not declared in this scope

       timer_group_intr_disable(_timerGroup, (_timerIndex == 0) ? TIMER_INTR_T0 : TIMER_INTR_T1);

                                                                                  ^

C:\Users\IDROID\Documents\Arduino\libraries\ESP32TimerInterrupt\src/ESP32TimerInterrupt.hpp: In member function 'void ESP32TimerInterrupt::reattachInterrupt()':

C:\Users\IDROID\Documents\Arduino\libraries\ESP32TimerInterrupt\src/ESP32TimerInterrupt.hpp:416:65: error: 'TIMER_INTR_T0' was not declared in this scope

       timer_group_intr_enable(_timerGroup, (_timerIndex == 0) ? TIMER_INTR_T0 : TIMER_INTR_T1);

                                                                 ^

C:\Users\IDROID\Documents\Arduino\libraries\ESP32TimerInterrupt\src/ESP32TimerInterrupt.hpp:416:81: error: 'TIMER_INTR_T1' was not declared in this scope

       timer_group_intr_enable(_timerGroup, (_timerIndex == 0) ? TIMER_INTR_T0 : TIMER_INTR_T1);

                                                                                 ^

C:\Users\IDROID\Documents\Arduino\libraries\ESP32TimerInterrupt\src/ESP32TimerInterrupt.hpp: In member function 'void ESP32TimerInterrupt::enableTimer()':

C:\Users\IDROID\Documents\Arduino\libraries\ESP32TimerInterrupt\src/ESP32TimerInterrupt.hpp:426:65: error: 'TIMER_INTR_T0' was not declared in this scope

       timer_group_intr_enable(_timerGroup, (_timerIndex == 0) ? TIMER_INTR_T0 : TIMER_INTR_T1);

                                                                 ^

C:\Users\IDROID\Documents\Arduino\libraries\ESP32TimerInterrupt\src/ESP32TimerInterrupt.hpp:426:81: error: 'TIMER_INTR_T1' was not declared in this scope

       timer_group_intr_enable(_timerGroup, (_timerIndex == 0) ? TIMER_INTR_T0 : TIMER_INTR_T1);

                                                                                 ^

exit status 1

Error compilando para la tarjeta ESP32 Dev Module.

**Comment by khoih-prog** [comment]
Hi @m66rob 

It's always better and saving time to scan thru **previous issues** as well as [**README**](https://github.com/khoih-prog/ESP32TimerInterrupt/blob/master/README.md) when got into an unsolved issue. 

For example, check either

1. [**TIMER_INTR_0 Not defined** #23](https://github.com/khoih-prog/ESP32TimerInterrupt/issues/23), 
2. [**Prerequisites**](https://github.com/khoih-prog/ESP32TimerInterrupt#prerequisites)

then you know you'll have to update to **ESP32 core v2.0.4**
